### PR TITLE
Event server keyboard events dont work for keys that have long press defined

### PR DESCRIPTION
**Pull request by HenrikJohnson**
## Description
Event server keyboard events don't send key down and key up events like a real keyboard so any key that is defined as having a long press option on it currently do not work to send through the event server.

The very simple fix I made is that if the key event comes from the event server don't process long press events at all.

## Motivation and Context
You can not send events for some keys for instance like select (enter) through the event server using the keyboard event type.

## How Has This Been Tested?
I have an event client that stopped working when I upgraded to Leia with the keyboard events that have long press options. With this change it now works again.

(Here is the source for my client https://github.com/HenrikJohnson/MauritzRemote/tree/master/RemoteServer)

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by HenrikJohnson** [comment]
The tests that failed seemed to have failed due to something on the OSX build environment unrelated to my change.

**Comment by notspiff** [comment]
makes sense to me.

**Comment by Rechi** [comment]
jenkins build this please

**Comment by garbear** [comment]
It looks like change has minimal impact, so if it fixes something, go for it.

**Comment by HenrikJohnson** [comment]
I am currently experiencing this issue in Leia so would probably be good for a back port as well if we are doing another patch on that branch.

**Comment by HenrikJohnson** [comment]
Any chance of getting this merged sometime soon?

**Comment by mglae** [comment]
+1 This is a fix for #15772 

@HenrikJohnson key.SetFromService(true) is missing for the UNICODE path:
```
--- a/xbmc/input/InputManager.cpp
+++ b/xbmc/input/InputManager.cpp
@@ -234,7 +234,8 @@ bool CInputManager::ProcessEventServer(int windowId, float frameTime)
       if (wKeyID & ES_FLAG_UNICODE)
       {
         key = CKey(0u, 0u, static_cast<wchar_t>(wKeyID & ~ES_FLAG_UNICODE), 0, 0, 0, 0);
+        key.SetFromService(true);
         return OnKey(key);
       }
 
       if (wKeyID == KEY_BUTTON_LEFT_ANALOG_TRIGGER)
```

**Comment by HenrikJohnson** [comment]
Will incorporate your change as well.

**Comment by HenrikJohnson** [comment]
So I guess merging from master is a no-no here. Does anybody have a suggestion on how I fix this or should I just close it and start a new one?

**Comment by wsnipex** [comment]
please get rid of the merge commit by rebasing on master and squashing to a single commit. Then just force push to your branch

**Comment by RomanSmolka** [comment]
@HenrikJohnson Any chance you do as wsnipex suggested to get this merged? The issue has been bugging me for a while too.

File: xbmc/input/InputManager.cpp
Comment (garbear) [inline xbmc/input/InputManager.cpp]: Grammar fixes, and wrap at 80 chars for small screens:

```c++
// Event server keyboard doesn't give normal key up and key down, so don't
// process for long press if that is the source
```

**Review by garbear — APPROVED** [review]
Approved once the cosmetic is addressed

**Comment by phunkyfish** [comment]
@HenrikJohnson can you clean this up so we can get it merged?

**Comment by dvershinin** [comment]
Works fine, should be merged.

**Comment by phunkyfish** [comment]
@dvershinin let's give @HenrikJohnson a few more days to make the cosmetic changes. 

If he's does not by then please ping me here and I will try to make them assuming he has allow maintainer edits enabled.

**Comment by phunkyfish** [comment]
Closed in favour of: https://github.com/xbmc/xbmc/pull/17806

**Pull request closed without merging**
